http/user/handlers.go

```diff
@@ -0,0 +1,191 @@
+package user
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"platform-go-challenge/internal/app/assets"
+	"platform-go-challenge/internal/app/users"
+	"platform-go-challenge/internal/pagination"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/gorilla/mux"
+)
+
+type Handler struct {
+	service       *users.Service
+	authorization string
+}
+
+func NewUserHandler(userService *users.Service, authorization string) Handler {
+	return Handler{
+		service:       userService,
+		authorization: authorization,
+	}
+}
+
+type starAction struct {
+	AssetID     uint32           `json:"asset_id"`
+	AssetType   assets.AssetType `json:"asset_type"`
+	Description string           `json:"description"`
+}
+
+func (s *starAction) DecodeAndValidate(r *http.Request) error {
+	if err := json.NewDecoder(r.Body).Decode(&s); err != nil {
+		return err
+	}
+	if _, found := assets.AssetTypes[s.AssetType]; !found {
+		return fmt.Errorf("invalid asset type: %v ", s.AssetType)
+	}
+	return nil
+}
+
+// GetDashboard
+func (h *Handler) GetDashboard(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-type", "application/json")
+
+	userID, err := strconv.Atoi(mux.Vars(r)["user_id"])
```

Comment:
Would be possible to send arbitrary userIds from the client and the app respond with the corresponding user data?

Reply:
Yes. This endpoint is not protected whatsoever. I only added some kind of protection to "write" endpoints but I guess I should have done the same with this one.